assets/js/hooks/useActivateModuleCallback.test.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * `useActivateModuleCallback` hook tests.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import {
+	createTestRegistry,
+	provideModules,
+	provideModuleRegistrations,
+	provideSiteInfo,
+	provideUserCapabilities,
+	renderHook,
+} from '../../../tests/js/test-utils';
+import { mockLocation } from '../../../tests/js/mock-browser-utils';
+import * as tracking from '../util/tracking';
+import {
+	CORE_USER,
+	PERMISSION_MANAGE_OPTIONS,
+} from '../googlesitekit/datastore/user/constants';
+import { CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+import { CORE_SITE } from '../googlesitekit/datastore/site/constants';
+import useActivateModuleCallback from './useActivateModuleCallback';
+
+const mockTrackEvent = jest.spyOn( tracking, 'trackEvent' );
+mockTrackEvent.mockImplementation( () => Promise.resolve() );
+
+describe( 'useActivateModuleCallback', () => {
+	mockLocation();
+	let registry;
+
+	beforeEach( () => {
+		jest.resetAllMocks();
```

Comment:
Generally I find this to be a matter of preference - in this case, I followed the prior art of creating a `mockTrackEvent` at the outer scope as I saw elsewhere in https://github.com/google/site-kit-wp/blob/480e31fa82bc799f229282672396c4b056d87468/assets/js/components/TourTooltips.test.js#L67

In that case, the mock is explicitly cleared before the specific tests it's used in:

https://github.com/google/site-kit-wp/blob/480e31fa82bc799f229282672396c4b056d87468/assets/js/components/TourTooltips.test.js#L247-L248

Arguably the use of `resetAllMocks` is a bit lazy, on the other hand it can be seen as a safe catchall, however it does introduce a marginal few extra CPU cycles. I'd be happy to move the `mockTrackEvent` creation inside the `beforeEach` and get rid of `resetAllMocks`.

Reply:
Actually - now I've tried it out, I've realised that even if we move the `mockTrackEvent = jest.spyOn( tracking, 'trackEvent' )` within the top-level `beforeEach`, the mock _still_ needs to be reset before it's used, as it seems like Jest is caching the spy internally from the first time it's called.

This being the case, I've made a change, keeping the `mockTrackEvent` definition at the outer scope to be consistent with the use in `TourTooltips.test.js`, but clearing the mock at the start of the test where it's going to be used, as well as adding the extra assertion. The commit is https://github.com/google/site-kit-wp/pull/4862/commits/14f81189f44ec5b41ab3a740499aad74a497b0c2, please see what you think.

Reply:
Huh, fair enough! Also happy to do whatever has been done in the past in the repo as well—in this case I couldn't see any other calls to `resetAllMocks` but fair point on it not really being dangerous.

The new commit looks good though, and thanks for the explanation! 👍🏻